src/components/manager/DataTypeSelect.js

```diff
@@ -0,0 +1,50 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import { Select, Spin } from "antd";
+import PropTypes from "prop-types";
+import { useSelector } from "react-redux";
+
+const DataTypeSelect = ({value, workflows, onChange}) => {
+    const [selected, setSelected] = useState(value ?? undefined);
+    const servicesFetching = useSelector((state) => state.services.isFetchingAll);
+    const workflowsFetching = useSelector((state) => state.serviceWorkflows.isFetchingAll);
+
+    useEffect(() => {
+        setSelected(value);
+    }, [value]);
+
+    const onChangeInner = useCallback((newSelected) => {
+        if (!value) setSelected(newSelected);
+        if (onChange) {
+            onChange(newSelected);
+        }
+    }, [value, onChange, selected]);
+
+    const options = useMemo(() => {
+        if (Array.isArray(workflows)) {
+            const dataTypes = new Set(workflows.map((w) => w.data_type));
```

Comment:
it would be nice if we could render the options with the `label` for data type (could put the `id` of it in parentheses with monospace just for the `id`?)

Reply:
The label is not included in the workflow objects, I could get the `label` using the data-types store with the workflow's data-type `id`